src/HtmPixelization.cc

```diff
@@ -0,0 +1,261 @@
+/*
+ * LSST Data Management System
+ * Copyright 2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/// \file
+/// \brief This file contains the HtmPixelization class implementation.
+
+#include "lsst/sphgeom/HtmPixelization.h"
+
+#include "lsst/sphgeom/curve.h"
+#include "lsst/sphgeom/orientation.h"
+
+#include "PixelFinder.h"
+
+
+namespace lsst {
+namespace sphgeom {
+
+namespace {
+
+// Raw HTM root vertex coordinates. UnitVector3d objects are avoided so that
+// library clients cannot run into the static initialization order fiasco.
```

Comment:
If you did want to use `UnitVector3d` and guard against the fiasco, you could just define this as a static variable in a function and have the variable return a reference to it.  That'd make it guaranteed to be initialized on first use, though it might make it a little more verbose for callers to access.